Update to work with pcb2gcode 2.1.0

the-snowwhite opened the pull request:
Thanks for maintaining pcb2gcode.
Since no one else is doing it I though I would give a shot at giving the gui a refresh.
Just an initial fix so that the gui runs with latest version without errors.
I expect to attempt incorporating most of the newer features next. :-)

the-snowwhite committed:
reorder items in CommonTab chronology
and adjust height of tabs

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
Change --optimize from bool to double
Updates pcb2gcodeGUI to work with latest pcb2gcode(v2.1.0)

Signed-off-by: Michael Brown <producer@holotronic.dk>

eyal0 commented on mainwindow.cpp:
```diff
@@ -270,7 +269,7 @@ void MainWindow::fillOutlineEnable(bool enable)
 void MainWindow::bridgesAvailable()
 {
-    bool bridgesEnabled = ui->vectorialCheckBox->isChecked() || ui->optimiseCheckBox->isChecked();
+    bool bridgesEnabled = ui->vectorialCheckBox->isChecked() || true;
```
Or true?

the-snowwhite replied:
The documentation is a bit confusing
from https://github.com/pcb2gcode/pcb2gcode/wiki/Manual

    --bridges unit
    add bridges with the given width to the outline cut. --bridgesnum bridges will be created for each outline 
    closed line. This option requires --optimise

I'm in doubt if the bridges generating still needs --optimize 
(or --vectorial for that matter)
If only --vectorial is required I can just delete the:

     || true

the-snowwhite replied:
if --optimize still is required in pcb2gcode I'll have to figure out testing for a non zero double instead of the or true

eyal0 replied:
That text is old and I didn't write it. Bridges doesn't require vectorial. Vectorial is mandatory anyway. Bridges doesn't require optimise, either.

The number is how many Bridges to put on each outline cut. The width is how wide each bridge should be. Bridges are places where the outline cut is modified so that it doesn't fully disconnect your board from the PCB. Sort of like making a perforation so that you can disconnect your board from the PCB when you are done.

I hope that makes sense! I'll look over the pcb2gcode help text at some point and fix it up. The man page will need work, too, though maintaining both the man page and the wiki has been a pain and I'd rather just maintain the wiki because I can add graphics and stuff there.

eyal0 commented:
I'm thrilled to have someome cover this. I don't use the GUI nor have time to work on it.

Any assistance that I can provide, let me know.

the-snowwhite commented:
Somehow my PR got spaced over 2 commits, so I have just:
squashed the 2 commits to 1
removed the || true (or true) you commented on.
and force pushed the branch

eyal0 commented on mainwindow.ui:
```diff
@@ -408,6 +408,19 @@
+           <item row="8" column="0">
+            <widget class="QLabel" name="optimiseLabel">
+             <property name="toolTip">
+              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;--optimise&lt;/span&gt;&lt;/p&gt;&lt;p&gt;optimise output gcode, reducing its output size up to 95%, while accepting a very little loss of precision&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Where did you get this text? It looks very similar to the help text except it is much higher a percentage, right?

the-snowwhite replied:
Its the original from the repo:
https://github.com/pcb2gcode/pcb2gcodeGUI/search?q=optimise+output+gcode%2C+reducing+its+output+size+up+to+95%25%2C+while+accepting+a+very+little+loss+of+precision&unscoped_q=optimise+output+gcode%2C+reducing+its+output+size+up+to+95%25%2C+while+accepting+a+very+little+loss+of+precision

I'll change it to 40 now

eyal0 commented on mainwindow.ui:
```diff
@@ -481,16 +494,31 @@
+              <double>0.000010000000000</double>
+             </property>
+             <property name="value">
+              <double>0.000½00000000000</double>
```
The value here is strange. Is that valid?

the-snowwhite replied:
A typo ½ and 1 are right beside each other on my keyboard. It's gone now

the-snowwhite committed:
Open and save in qt-creator(4.11.1)

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
Change obsoleted --dpi to --mirror-axis

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
reorder items in OutlineTab chronology
and change obsoleted --outline-width to --cut-vertfeed

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
Remove --vectorial as its now mandatory

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
Remove commented out code

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite commented:
With vectorial now mandatory I changed the --dpi parameter to --mirror-axis resolving
this issue: #16
exchanged the obsoleted --outline-width to --cut-vertfeed and moved it higher up
Lastly removed all the --vectorial stuff as Its default true
Did some enhancing of the gui layout underway. #12

the-snowwhite committed:
Change --optimise tooltip percentage to a more realistic one

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite committed:
App now remembers last folder of Gerber Pre/Post-ambles output and config files separately

Signed-off-by: Michael Brown <producer@holotronic.dk>

the-snowwhite commented:
@eyal0  thanks for the invite.
So what are the rules here about merging ?
I'm used to not merging your own PR unless in an emergency :-)

eyal0 commented:
We don't have any rules here, I think. It's just me maintaining stuff nowadays.

The pcb2gcode repo has unit tests and coverage. If you want me to set this project up with those, I can help. I'm no expert at GUI, though, so I'm not sure how that works.

As far as I'm concerned, you can merge your own stuff.
